### PR TITLE
remove forced voicemail exchange

### PR DESCRIPTION
**Pull request by akhrail1996**

### CODE REVIEW

File: lib/t/calls.ex [hunk 1]
```diff
@@ -322,42 +322,33 @@ defmodule T.Calls do
     Multi.new()
     |> Multi.run(:match, fn _repo, _changes ->
```
Comment (ruslandoga) [inline lib/t/calls.ex:323]: before I added the new timer logic, this step was called `:mate`, maybe it makes sense to name it that again?

File: lib/t/calls.ex [hunk 2]
```diff
@@ -322,42 +322,33 @@ defmodule T.Calls do
...
       match =
-        Matches.matches_with_undying_events_q()
+        Match
         |> where(id: ^match_id)
         |> where([m], m.user_id_1 == ^caller_id or m.user_id_2 == ^caller_id)
-        |> select([m, undying_event: e], {m.inserted_at, [m.user_id_1, m.user_id_2], e.timestamp})
+        |> select([m], {[m.user_id_1, m.user_id_2]})
```
Comment (ruslandoga) [inline lib/t/calls.ex:328]:
```suggestion
        |> select([m], [m.user_id_1, m.user_id_2])
```

File: lib/t/calls.ex [hunk 3]
```diff
@@ -322,42 +322,33 @@ defmodule T.Calls do
...
         |> Repo.one()
 
       case match do
-        {inserted_at, users, undying_event_at} ->
+        {users} ->
```
Comment (ruslandoga) [inline lib/t/calls.ex:332]:
```suggestion
        users ->
```

Comment (ruslandoga) [reply lib/t/calls.ex:332]: I think it might have been simpler before I added the timer logic, I'll check it out tomorrow.

File: lib/t/calls.ex [hunk 4]
```diff
@@ -322,42 +322,33 @@ defmodule T.Calls do
...
-        |> Repo.update_all(set: [exchanged_voicemail: true])
 
-        {:ok, true}
-      else
-        {:ok, false}
-      end
+      {:ok, s3_keys}
     end)
     |> Multi.insert(:voicemail, voicemail)
     |> Oban.insert(:push, fn %{match: %{mate: mate}} ->
```
Comment (ruslandoga) [inline lib/t/calls.ex:354]:
```suggestion
    |> Oban.insert(:push, fn :%{mate: mate} ->
```

File: lib/t/calls.ex [hunk 5]
```diff
@@ -384,18 +374,9 @@ defmodule T.Calls do
...
-          unless undying_event_at do
-            if exchanged_voicemail do
-              inserted_at
-              |> DateTime.from_naive!("Etc/UTC")
-              |> DateTime.add(Matches.match_ttl())
-            end
-          end
-
-        broadcast_payload = %{voicemail: voicemail, expiration_date: new_expiration_date}
+        broadcast_payload = %{voicemail: voicemail}
```
Comment (ruslandoga) [inline lib/t/calls.ex:377]:
```suggestion
        voicemail # no need to construct a new payload since there is no value other than voicemail in it
```

File: lib/t/matches.ex [hunk 6]
```diff
@@ -326,29 +322,23 @@ defmodule T.Matches do
...
     |> Enum.map(fn {match, undying_event_timestamp} ->
       expiration_date =
-        cond do
-          undying_event_timestamp -> nil
-          match.exchanged_voicemail -> post_voicemail_expiration_date(match)
-          true -> pre_voicemail_expiration_date(match)
+        if undying_event_timestamp do
+          nil
+        else
+          expiration_date(match)
```
Comment (ruslandoga) [inline lib/t/matches.ex:328]:
```suggestion
unless undying_event_timestamp do
  expiration_date(match)
end
```

File: lib/t/matches/match.ex [hunk 7]
```diff
@@ -8,8 +8,6 @@ defmodule T.Matches.Match do
     field :user_id_1, Ecto.Bigflake.UUID
     field :user_id_2, Ecto.Bigflake.UUID
 
-    field :exchanged_voicemail, :boolean
```
Comment (ruslandoga) [inline lib/t/matches/match.ex:11]: this seems like a breaking change, would old client be ok with this field removed from responses?

File: lib/t_web/channels/feed_channel.ex [hunk 8]
```diff
@@ -424,8 +421,7 @@ defmodule TWeb.FeedChannel do
             profile: profile,
             screen_width: screen_width,
             inserted_at: inserted_at,
-            expiration_date: expiration_date,
-            exchanged_voice: false
```
Comment (ruslandoga) [inline lib/t_web/channels/feed_channel.ex:428]: or rather this is could be the breaking change, since the client doesn't care about our DB but it does care about the immediate responses from backend

Comment (ruslandoga) [reply lib/t_web/channels/feed_channel.ex:428]: is it ok to remove this two fields? how does the client handle their absence? or maybe it doesn't use them at all?

File: lib/t_web/channels/feed_channel.ex [hunk 9]
```diff
@@ -517,7 +513,7 @@ defmodule TWeb.FeedChannel do
   end
 
   def handle_info({Calls, [:voicemail, :received], payload}, socket) do
-    %{voicemail: voicemail, expiration_date: expiration_date} = payload
+    %{voicemail: voicemail} = payload
```
Comment (ruslandoga) [inline lib/t_web/channels/feed_channel.ex:516]:
```suggestion
   voicemail # if the suggestions from above are implemented
```

File: test/t/calls/voicemail_test.exs [hunk 10]
```diff
@@ -31,32 +31,27 @@ defmodule T.Calls.VoicemailTest do
                Calls.voicemail_save_message(caller.id, match.id, s3_key)
     end
 
-    test "success: saves/overwrites voicemail, updates match.exchanged_voicemail" do
+    test "success: saves/overwrites voicemail" do
```
Comment (ruslandoga) [inline test/t/calls/voicemail_test.exs:34]: this test and many others are redundant as they don't need to do the exchanges anymore to check for `exchanged_voicemail`

File: test/t/matches/expired_match_test.exs [hunk 11]
```diff
@@ -22,50 +22,10 @@ defmodule T.Matches.ExpiredMatchTest do
...
   describe "expiration_list_soon_to_expire/0,1" do
-    defp in_pre_voicemail_exp_notification_window(match) do
-      match.inserted_at
-      |> DateTime.from_naive!("Etc/UTC")
-      |> DateTime.add(Matches.pre_voicemail_ttl())
-      |> DateTime.add(_3h = -3 * 3600)
-    end
-
-    defp in_post_voicemail_exp_notification_window(match) do
+    defp exp_notification_window(match) do
```
Comment (ruslandoga) [inline test/t/matches/expired_match_test.exs:28]:
```suggestion
    defp in_exp_notification_window(match) do
```

Comment (ruslandoga) [reply test/t/matches/expired_match_test.exs:28]: since the function returns a single value in the window, not the window itself, which would be a range

File: test/t_web/channels/feed_channel_test.exs [hunk 12]
```diff
@@ -390,7 +379,6 @@ defmodule TWeb.FeedChannelTest do
       assert reply == %{
                "mode" => "normal",
                "since_live_date" => since_live_date,
-               "match_expiration_duration" => 172_800 = Matches.pre_voicemail_ttl(),
```
Comment (ruslandoga) [inline test/t_web/channels/feed_channel_test.exs:393]: just a note, I know this field needs to be removed but I believe it's best to make these changes in separate PRs otherwise it's easy to overlook ...

**Comment by ruslandoga** [comment]
Would it be ok for me to push some commits on top in the morning? After that I think it would be ok to merge

File: lib/t/calls.ex (hunk 3 above)
Comment (ruslandoga) [reply lib/t/calls.ex:332]: https://github.com/getsince/test3/pull/479/commits/27d5d2ed0529d00ae0ec1615a82b3fe47a1b6554

File: lib/t/matches/match.ex (hunk 7 above)
Comment (akhrail1996) [reply lib/t/matches/match.ex:11]: with this in particular - yes (https://github.com/getsince/test5/pull/202/files#diff-d5a373f3ff78460009f5de32b1aa475744f13f1125990d4de348896dc7c3bc42L36);

with one other - no (https://github.com/getsince/test5/pull/202/files#diff-ff2ea48ff2f5f39ca8c03130d73e2dcb1e534ad449e4020fa77ec29657958c17L189)
-> I will revert this one so there is no breaking change

File: lib/t_web/channels/feed_channel.ex (hunk 8 above)
Comment (akhrail1996) [reply lib/t_web/channels/feed_channel.ex:428]: only `exchanged_voice` removed here, yeah it's okay

File: test/t/calls/voicemail_test.exs (hunk 10 above)
Comment (akhrail1996) [reply test/t/calls/voicemail_test.exs:34]: it checks saves and overwrites, ain't it?

**Comment by akhrail1996** [comment]
@ruslandoga sure
